Framework/Crystal/inc/MantidCrystal/FilterPeaks.h

```diff
@@ -71,6 +72,33 @@ class MANTID_CRYSTAL_DLL FilterPeaks : public API::Algorithm {
         filteredWS.addPeak(currentPeak);
     }
   }
+
+  /**
+   * @brief Select peaks from the input peak workspace by checking a string
+   *        value
+   *
+   * @tparam Comparator
+   * @param inputWS :: the input peaks workspace containing peaks to be filtered
+   * @param filteredWS :: the output peaks workspace which will contain a subset
+   * of the inputWS
+   * @param filterFunction :: function extracting the value to filter on from
+   * the peak object
+   * @param filterValue :: the string value to check
+   */
+  template <typename Comparator>
+  void filterPeaksStr(const Mantid::API::IPeaksWorkspace &inputWS,
```

Comment:
`Comparator` is meant to be a function, not a type. To me, the following function signature seems more appropriate:
```C++
template <typename Comparator>
void filterPeaksStr(const Mantid::API::IPeaksWorkspace &inputWS,
                    Mantid::API::IPeaksWorkspace &filteredWS,
                    const FilterFunctionStr &filterFunction,
                    const std::string filterValue,
                    Comparator operatorFunc);
```
Later you could do:
```C++
filterPeaksStr(*inputWS, *selectedWS, filterFunction, bankname, std::equal_to<std::string>());
```
The advantage of this is that you could pass custom Comparator functions.
See [here for a nice working example](https://godbolt.org/z/3pwX4G).

Reply:
The current style in use
`filterPeaks<std::less<double>>(*inputWS, *filteredWS, filterFunction,
                                     filterValue);` is already decided by the previous developer, and I am only here to extending its supported type from double to string.
I would prefer not include style change as it is beyond the scope of this PR.

Reply:
I guess that's fair 😃